### PR TITLE
Export visit utils

### PR DESCRIPTION
**Pull request by nkimaina**

### CODE REVIEW

**Comment by nkimaina** [comment]
Need to use the exported functionality in a 3rd party esm module

**Review by donaldkibet — APPROVED** [review]
LTGM. Should we drop the auto generated translation and manage them manually

**Review by FlorianRappl — COMMENTED** [review]
LGTM with the exception of the translation file.

We should prefer real keys to fallbacks as keys.

That means instead of 

```
"Please enter a date that is either on or before today": "Please enter a date that is either on or before today",
```

we'd define

```
"errorFutureDate": "Please enter a date that is either on or before today",
```

**Comment by nkimaina** [comment]
@FlorianRappl There's a tool that runs to build the translations json, and that's what it did when I committed my changes, I wasn't responsible for coming up with the translations. That said, I tried to change it and about 8 tests broke. On investigating, it seems to injecting the keys to the DOM, as readable text. Am I missing something? do we try to handle this issue in a separate PR?

**Comment by FlorianRappl** [comment]
@nkimaina I know, but we already started removing this tool in other repositories. We should do it here, too. @brandones can give more insights!

Sure we could also do it in another PR, I'd just love to get a good grip on localizations now, because from experience it will only get harder later.

**Review by FlorianRappl — APPROVED** [review]

**Comment by nkimaina** [comment]
No worries @FlorianRappl. Let's wait for @brandones to tell us more about what is required to get it removed because I have the mappings done correct locally.

**Comment by brandones** [comment]
You can just remove it -- there's nothing to be done other than to be careful when developing and reviewing code that translation keys get added to and removed from `en.json` as necessary. But hopefully we're doing that already :)

File: src/widgets/visit/start-visit-prompt.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { newModalItem } from "./visit-dialog-resource";
+import { startVisitConfirmation } from "./visit-button-component";
+
+export function startVisitPrompt(onPromptClosed?: Function) {
```
Comment (FlorianRappl) [inline src/widgets/visit/start-visit-prompt.ts]: For the `Function` just use `() => void`

File: src/widgets/visit/start-visit-prompt.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  newModalItem({
+    component: startVisitConfirmation(),
+    name: "Prompt start Visit",
+    props: { closeComponent: () => onPromptClosed && onPromptClosed() }
```
Comment (FlorianRappl) [inline src/widgets/visit/start-visit-prompt.ts]: Alternatively you can just `onPromptClosed ?.()`

**Review by FlorianRappl — COMMENTED** [review]
Besides the few smaller cosmetic things I'd say great work!

**Review by FlorianRappl — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**